### PR TITLE
[incubator/traefik] Create Incubator Chart for Traefik

### PR DESCRIPTION
**Pull request by dtomcej**
#### What this PR does / why we need it:

This PR creates the incubator chart for use with upcoming Traefik v2 release.

##### Technical requirements
- [x] All Chart dependencies should also be submitted independently
- [x] Must pass the linter (helm lint)
- [ ] Must successfully launch with default values (helm install .)
    - [ ] All pods go to the running state
    - [ ] All services have at least one endpoint
- [ ] Must include source GitHub repositories for images used in the Chart
- [ ] Images should not have any major security vulnerabilities
- [ ] Must be up-to-date with the latest stable Helm/Kubernetes features
    - [x] Use Deployments in favor of ReplicationControllers
- [ ] Should follow Kubernetes best practices
    - [x] Include Health Checks wherever practical
    - [ ] Allow configurable resource requests and limits
- [ ] Provide a method for data persistence (if applicable)
- [ ] Support application upgrades
- [ ] Allow customization of the application configuration
- [ ] Provide a secure default configuration
- [ ] Do not leverage alpha features of Kubernetes
- [ ] Includes a NOTES.txt explaining how to use the application after install
- [ ] Follows best practices (especially for labels and values)
##### Documentation requirements
- [ ] Must include an in-depth README.md, including:
    - [ ] Short description of the Chart
    - [ ] Any prerequisites or requirements
    - [ ] Customization: explaining options in values.yaml and their defaults
- [ ] Must include a short NOTES.txt, including:
    - [ ] Any relevant post-installation information for the Chart
    - [ ] Instructions on how to access the application or service provided by the Chart

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md

### CODE REVIEW

File: incubator/traefik/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+apiVersion: v1
+name: traefik
+version: 2.0.0-alpha1
```
Comment (paulczar) [inline incubator/traefik/Chart.yaml:3]: chart version should not necessarily match app version.   Given this is in incubation, I would set `version: 0.0.1`

**Review by paulczar — CHANGES_REQUESTED** [review]

File: incubator/traefik/Chart.yaml (hunk 1 above)
Comment (dtomcej) [reply incubator/traefik/Chart.yaml:3]: The issue with that is that we already have `stable/traefik` that corresponds to the v1.X version of the app.

We wanted to have this chart be the v2 version of the chart to correspond with the v2 of the app.

**Comment by dtomcej** [comment]
Bump.

**Comment by maorfr** [comment]
hey @dtomcej 

we already have a stable/traefik chart

Have you considered hosting this chart and adding it to helm hub?
https://github.com/helm/hub/blob/master/Repositories.md

**Comment by dtomcej** [comment]
Hello @maorfr,

>we already have a stable/traefik chart

yes! we already manage that one!

This PR creates the incubator chart. We have a new v2 for traefik that is completely different configuration wise, and we want to have an incubator to test with before moving it to stable!

**Comment by maorfr** [comment]
hey @dtomcej,

thanks for your response.
we recently wanted to do the same for the Jenkins chart, but in a call with other maintainers it was decided that we should not host two versions for the same chart.

We believe that the best way to move forward with this is that you host the new chart yourself [[1](https://github.com/helm/hub/blob/master/Repositories.md)] and when it is all done you merge it directly to stable with a major version bump.

what do you think?

With Jenkins, we ended up not having an incubator version, and did all the changes directly to stable, with a major version bump.

**Pull request closed without merging**